### ISSUE TITLE
Message not infomative when names is omitted

### ISSUE THREAD

**Issue by graemeblair** [DeclareDesign/fabricatr#11]
```set.seed(2)
fabricate(N=2, 1:N)
## Error in fabricate(N = 2, 1:N): object 'N' not found
```

N should be found; issue is what to do with 1:N

**Comment by graemeblair** [comment]
The problem is 1:N is being taken as the data argument, so it's about ordering ...

**Comment by aaronrudkin** [comment]
I thought a little bit about this. I don't think there's much use in going down the rabbit hole trying to predict user behaviour where they provide arguments that do not have explicitly names and also are not detected by argument order; it is totally unclear what the user intends to provide with the 1:N in this example -- are they providing a malformed data frame (error out either because the data frame is malformed or because both N and data are specified), or are they attempting to create a variable that they have not explicitly named?

What I think is useful is ensuring that the user gets an error that allows them to determine what we think they passed us. I've updated fabricate (commit forthcoming), to provide the following error:

`          stop(
            "The data argument must be a data object. The argument call, ", deparse(substitute(data)), ", was not a data object (e.g. a data.frame, tibble, sf object, or convertible matrix)."
          )`

e.g. " The data argument must be a data object. The argument call, 1:N, was not a data object (e.g. a data.frame, tibble, sf object, or convertible matrix)."

This should allow the user to understand that we are interpreting the first non-explicitly provided value as an object being passed to the data argument.

In theory if the user's unspecified call ends up being a valid data frame, then the code will error later when it realizes in the fabricate_data_single_level function that the user has provided an N and a data variable; I have also modified this error to output what fabricatr thinks the provided N and data values are -- it's a little ugly because `deparse(substitute(data))` will produce "data=data" while `deparse(data)` will produce a deparsed version of the data frame, but I suspect it'll still be enough to tip off a user for now. Perhaps someone with more NSE experience than me can suggest ways of making this look better.

Example for something that's not a valid data call:
`fabricate(N=2, 1:N)
`

Example for something that is a valid data call:
`fabricate(N=2, data.frame(test1=c(1, 2), test2=c(3, 4)))
`

These changes are in the most recent commit to the feature branch I'm working on as of  7fb8045. If no one has further input I will close this tomorrow.
